File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.java
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.OutputStreamWriter;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        boolean[] sieve= new boolean[10001]; // 기본 false로 초기화 -> 0
        for (int i=1; i<10001; i++){
            if (sieve[i]) continue;
            int next = i;
            while (true) {
                // now와 mod를 int 형으로 초기화하는데
//                now는 next의 값을 mod는 선언만
                int now = next, mod;
                while (true) {
                    mod = now %10;
                    now /= 10;
                    next += mod;
                    if (now == 0) break;
                }
                if (next > 10000) break;
                sieve[next] = true;
            }
        }
        for (int i = 1; i< 10001; i++){
            if (!sieve[i]) {
                bw.write(i+"\n");
            }
        }

        bw.close();
    }

}

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.py
sieve = [0] * 10001

for i in range(1,10001):
    if sieve[i] : continue
    next = i
    while 1:
        now = next
        while 1:
            now, mod = divmod(now,10)
            next += mod
            if now == 0: break
        if next > 10000: break
        sieve[next] = 1

for i in range(1,10001):
    if not sieve[i]:
        print(i)